static/app/views/starfish/index.tsx

```diff
@@ -16,7 +19,58 @@ type Props = {
   organization: Organization;
 };
 
-const queryClient = new QueryClient();
+// const shouldCacheQuery = (query: Query) => {
+//   const statsPeriod = query.queryKey?.[1]?.statsPeriod;
+//   const data: any = query.state.data;
+//   const statusCode = data[2]?.statusCode;
+//   return statsPeriod === '7d' && statusCode === 200;
+// };
+
+// const p = queryPersis;
+// queryCache.subscribe(({query, type}) => {
+//   const key = query.queryKey;
+//   const data = query.state.data;
+//   console.log(key);
+//   // console.log(key, statsPeriod);
+// });
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      cacheTime: 1000 * 60 * 60, // 1 hour
+    },
+  },
+});
+
+const persister = createSyncStoragePersister({
+  storage: window.localStorage,
+  throttleTime: 5000,
+  serialize: res => {
+    const queries = res.clientState.queries;
+    queries.forEach(query => {
+      const state = query.state as QueryState & {pageLinks?: string};
+      const data = state.data as [any, string, ResponseMeta];
+      const pageLinks = data?.[2]?.getResponseHeader('Link') ?? undefined;
+      state.pageLinks = pageLinks;
+    });
+    return JSON.stringify(res);
+  },
+  deserialize: res => {
+    const result = JSON.parse(res);
+    const queries = result.clientState.queries;
+    queries.forEach(query => {
+      const state: QueryState & {pageLinks?: string} = query.state;
+      const data = state.data as [any, string, ResponseMeta];
+      const pageLinks = state.pageLinks ?? null;
+      data[2].getResponseHeader = (headerName: string) => {
+        if (headerName === 'Link') {
+          return pageLinks;
+        }
+        return null;
+      };
```

Comment:
This is a bit hacky, but it does allow us to retain the original api to get the response header. lmk if anyone has thoughts on this